client-app/core/composables/useErrorsTranslator.test.ts

```diff
@@ -0,0 +1,267 @@
+import { describe, expect, it, vi } from "vitest";
+import { useErrorsTranslator } from "./useErrorsTranslator";
+
+function mockI18n(): void {
+  vi.mock("vue-i18n", async (importOriginal) => {
+    const actualI18n = await importOriginal();
+    return {
+      ...{ actualI18n },
+      useI18n: vi.fn().mockReturnValue({
+        t: (key: string) => key,
+        te: (key: string) => key,
+      }),
+    };
+  });
+}
+
+describe("use-errors-translator", () => {
+  mockI18n();
+
+  const { localizedItemsErrors, setErrors } = useErrorsTranslator("validation_error");
+
+  it("Test for unexisting error", () => {
+    setErrors([
+      {
+        errorCode: "PRODUCT_QTY_INSUFFICIENT",
+        objectId: "78e7a44e292a439a9a4e059bda5a398f",
+        objectType: "CartProduct",
+      },
+    ]);
+
+    expect(localizedItemsErrors.value["123"]).toBeUndefined();
+  });
+
+  it("Test for insufficient product quantity", () => {
+    setErrors([
+      {
+        errorCode: "PRODUCT_QTY_INSUFFICIENT",
+        errorParameters: [
+          {
+            key: "new_qty",
+            value: 1600,
+          },
+          {
+            key: "avail_qty",
+            value: 1586,
+          },
+        ],
+        objectId: "78e7a44e292a439a9a4e059bda5a398f",
+        objectType: "CartProduct",
+      },
+    ]);
+
+    expect(localizedItemsErrors.value["78e7a44e292a439a9a4e059bda5a398f"]).toBeDefined();
+    expect(
+      localizedItemsErrors.value["78e7a44e292a439a9a4e059bda5a398f"].includes(
+        "validation_error.PRODUCT_QTY_INSUFFICIENT",
+      ),
+    ).toBeTruthy();
+  });
+
+  it("Test for product quantity changed error", () => {
```

Comment:
no needs to test different but similar by structure sets of data. We are testing logic, this logic is deterministic.
It could have made sense if you did check error parameters for example

Reply:
Actually we should test localized values, still searching how to add locale dictionary to tests

Reply:
we don't need to test i18n library logic, will be enough to mock `t` function and check if it has been called with correct key and parameters

```ts
const mocks = vi.hoisted(() => {
  return {
    t: vi.fn().mockImplementation((key: string, params: unknown) => key + JSON.stringify(params)),
  };
});

function mockI18n(): void {
  vi.mock("vue-i18n", () => {
    return {
      useI18n: vi.fn().mockReturnValue({
        t: mocks.t,
        te: (key: string) => key,
      }),
    };
  });
}

describe("use-errors-translator", () => {
  mockI18n();
  
  // ...
  
	it("should retrun insufficient product quantity error", () => {
	    const { localizedItemsErrors, setErrors } = useErrorsTranslator("validation_error");
	
	    setErrors([
	      {
	        errorCode: "PRODUCT_QTY_INSUFFICIENT",
	        errorParameters: [
	          {
	            key: "new_qty",
	            value: 1600,
	          },
	          {
	            key: "avail_qty",
	            value: 1586,
	          },
	        ],
	        objectId: "78e7a44e292a439a9a4e059bda5a398f",
	        objectType: "CartProduct",
	      },
	    ]);
	
	    expect(localizedItemsErrors.value["78e7a44e292a439a9a4e059bda5a398f"]).toBeDefined();
	    expect(mocks.t).toBeCalledWith("validation_error.PRODUCT_QTY_INSUFFICIENT", { avail_qty: 1586, new_qty: 1600 });
	
	    expect(localizedItemsErrors.value["78e7a44e292a439a9a4e059bda5a398f"]).toEqual([
	      'validation_error.PRODUCT_QTY_INSUFFICIENT{"new_qty":1600,"avail_qty":1586}',
	    ]);
	  });
```